src/Admin/AbstractAdmin.php

```diff
@@ -2312,11 +2336,38 @@ public function getChild($code)
     }
 
     /**
+     * NEXT_MAJOR: Uncomment the argument.
+     *
      * @final since sonata-project/admin-bundle 3.102.
      */
-    public function setParent(AdminInterface $parent)
+    public function setParent(AdminInterface $parent/*, string $parentAssociationMapping*/)
     {
         $this->parent = $parent;
+
+        // NEXT_MAJOR: Uncomment the following line
+        // $this->parentAssociationMapping[$parent->getCode()] = $parentAssociationMapping;
+
+        // NEXT_MAJOR: Remove next line.
+        $args = \func_get_args();
+
+        // NEXT_MAJOR: Remove this entire if-else block.
+        if (isset($args[1])) {
+            if (!\is_string($args[1])) {
+                throw new \TypeError(sprintf(
+                    'Argument 2 passed to "%s()" must be of the type string, "%s" given.',
+                    __METHOD__,
+                    \is_object($args[1]) ? \get_class($args[1]) : \gettype($args[1])
+                ));
+            }
+
+            $this->parentAssociationMapping[$parent->getCode()] = $args[1];
+        } else {
+            @trigger_error(sprintf(
+                'Not passing argument 2 to "%s()" is deprecated since'
```

Comment:
This `else` block will be reached when the argument is not provided or when passing `null` as its value. IMO, we should update this message to reflect that situation or replace the `isset()` call with `array_key_exists()` otherwise.

Reply:
Done with array_key_exists